rcl_logging_log4cxx/src/rcl_logging_log4cxx/rcl_logging_log4cxx.cpp

```diff
@@ -78,37 +68,37 @@ extern "C" {
 
 #include "rcl_logging_log4cxx/logging_interface.h"
 
-#define RC_LOGGING_RET_OK                          (0)
-#define RC_LOGGING_RET_WARN                        (1)
-#define RC_LOGGING_RET_ERROR                       (2)
-#define RC_LOGGING_RET_INVALID_ARGUMENT            (11)
-#define RC_LOGGING_RET_CONFIG_FILE_DOESNT_EXIST    (21)
-#define RC_LOGGING_RET_CONFIG_FILE_INVALID         (22)
+#define RCL_LOGGING_RET_OK                          (0)
+#define RCL_LOGGING_RET_ERROR                       (2)
```

Comment:
It seems odd that this is defined in a `cpp` file. What's the relation of the functions in this file with https://github.com/ros2/rcl/blob/master/rcl/include/rcl/logging_external_interface.h ? If the return type is [rcl_ret_t](https://github.com/ros2/rcl/blob/d655147310e3e63aff1555ad40fca1bb9cb7783e/rcl/include/rcl/logging_external_interface.h#L41), then it seems like the return codes should be in https://github.com/ros2/rcl/blob/master/rcl/include/rcl/types.h .

Reply:
Sigh, yes.  This entire package hasn't exactly been done in the correct way.

The relationship with `logging_external_interface.h` is that it *must* implement the exact same API.  But we have to essentially hand-duplicate that API interface here to break a cycle.  The `rcl` package depends on the default implementation (which we are currently switching to `rcl_logging_spdlog`), so the implementation cannot also depend on `rcl` in order to get access to the API header.

Probably the correct answer is to split that API header out of `rcl` and either into its own package or into a "common" API package here, but it fell off the radar.

Reply:
Roger. Is there a ticket for moving the API to a separate package?

Reply:
No, but that's a good idea.  I've just opened https://github.com/ros2/rcl_logging/issues/20 to track it.